### PR TITLE
BuildResidentialHPXML measure: Add optional validation argument (for speed)

### PR DESCRIPTION
**Pull request by shorowit**
## Pull Request Description

BuildResidentialHPXML measure: Adds an optional argument for whether the HPXML file is validated; defaults to false. Most workflows using the BuildResidentialHPXML measure are validating the HPXML twice (once at the end of BuildResidentialHPXML and once at the beginning of HPXMLtoOpenStudio), which incurs a performance penalty. We now only apply validation in the BuildResidentialHPXML measure in special circumstances (e.g., creating sample files and unit tests).

Verified that each ResStock simulation is now faster -- about 0.7s faster using run_analysis.rb locally w/ Linux.

## Checklist

Not all may apply:

- [ ] EPvalidator.xml has been updated
- [ ] Tests (and test files) have been updated
- [ ] Documentation has been updated
- [x] Changelog has been updated
- [x] `openstudio tasks.rb update_measures` has been run
- [x] No unexpected regression test changes on CI (checked comparison artifacts)

### CODE REVIEW

File: BuildResidentialHPXML/measure.rb [hunk 1]
```diff
@@ -2897,6 +2897,12 @@ def arguments(model)
       args << arg
     end
 
+    arg = OpenStudio::Measure::OSArgument::makeBoolArgument('apply_validation', false)
```
Comment (joseph-robertson) [inline BuildResidentialHPXML/measure.rb:2900]: Can we make `apply_defaults` consistent with this new `apply_validation` argument? I'm thinking in terms of position within the arguments section (i.e., at the bottom), as well as display name and description formatting.

**Review by joseph-robertson — CHANGES_REQUESTED** [review]

File: BuildResidentialHPXML/tests/build_residential_hpxml_test.rb [hunk 2]
```diff
@@ -339,6 +326,7 @@ def test_workflows
   def _set_measure_argument_values(hpxml_file, args)
     args['hpxml_path'] = "tests/extra_files/#{hpxml_file}"
     args['apply_defaults'] = true
```
Comment (joseph-robertson) [inline BuildResidentialHPXML/tests/build_residential_hpxml_test.rb:328]: Why should this be true?

File: BuildResidentialHPXML/measure.rb (hunk 1 above)
Comment (shorowit) [reply BuildResidentialHPXML/measure.rb:2900]: Done.

File: BuildResidentialHPXML/tests/build_residential_hpxml_test.rb (hunk 2 above)
Comment (shorowit) [reply BuildResidentialHPXML/tests/build_residential_hpxml_test.rb:328]: So that it's tested?

**Review by joseph-robertson — APPROVED** [review]

**Pull request merged**